cloud/google/config/configtemplate.go

```diff
@@ -147,10 +147,39 @@ spec:
           - name: GOOGLE_APPLICATION_CREDENTIALS
             value: /etc/credentials/service-account.json
         command:
-        - "./gce-machine-controller"
+        - "./gce-controller"
         args:
         - --kubeconfig=/etc/kubernetes/admin.conf
         - --machinesetup=/etc/machinesetup/machine_setup_configs.yaml
+        - --controller=machine
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+      - name: gce-cluster-controller
+        image: {{ .GCEControllerImage }}
+        volumeMounts:
+          - name: config
+            mountPath: /etc/kubernetes
+          - name: certs
+            mountPath: /etc/ssl/certs
+          - name: credentials
+            mountPath: /etc/credentials
+          - name: sshkeys
+            mountPath: /etc/sshkeys
+          - name: machine-setup
+            mountPath: /etc/machinesetup
```

Comment:
The way I was thinking about it, the machine controller would be responsible for making the machines into nodes, the cluster controller would just generate the machine yaml. That way any kubeadm calls can happen from the machine controller. Does that make sense to you?

Reply:
Yes -- makes complete sense. But is kubeadm actually mounted for the machine controller? it seems like it isn't, right? Do you just need to rebase ontop of the latest?

Reply:
It is, that line is hidden in the PR because I didn't anything close enough to it. You can see it [here](https://github.com/mkjelland/cluster-api/blob/6f57fb2ff841627e2ea88bef18bf4ed081388da9/cloud/google/config/configtemplate.go#L144)